File: backend/app/schemas/trade.py
from datetime import datetime
from enum import Enum

from pydantic import BaseModel, Field


class TradeType(str, Enum):
    buy = "BUY"
    sell = "SELL"


class TradeStatus(str, Enum):
    open = "open"
    closed = "closed"


class PaperTradeOpenRequest(BaseModel):
    symbol: str
    type: TradeType
    size: float = Field(gt=0)
    entry_price: float | None = Field(default=None, gt=0)
    stop_loss: float | None = Field(default=None, gt=0, description="Optional stop loss price")
    take_profit: float | None = Field(default=None, gt=0, description="Optional take profit price")


class TradeCloseRequest(BaseModel):
    exit_price: float | None = Field(default=None, gt=0)


class TradeResponse(BaseModel):
    id: str
    user_id: str
    symbol: str
    market_type: str
    type: TradeType
    entry_price: float
    exit_price: float | None = None
    stop_loss: float | None = None
    take_profit: float | None = None
    size: float
    status: TradeStatus
    is_paper: bool
    pnl: float | None = None
    pnl_percent: float | None = None
    created_at: datetime
    closed_at: datetime | None = None


class TradeListResponse(BaseModel):
    user_id: str
    trades: list[TradeResponse]
    open_count: int
    closed_count: int
    total_realized_pnl: float

File: frontend/src/types/trade.ts
export type TradeType = "BUY" | "SELL";
export type TradeStatus = "open" | "closed";

export interface Trade {
  id: string;
  user_id: string;
  symbol: string;
  market_type: string;
  type: TradeType;
  entry_price: number;
  exit_price?: number | null;
  stop_loss?: number | null;
  take_profit?: number | null;
  size: number;
  status: TradeStatus;
  is_paper: boolean;
  pnl?: number | null;
  pnl_percent?: number | null;
  created_at: string;
  closed_at?: string | null;
}

export interface TradeList {
  user_id: string;
  trades: Trade[];
  open_count: number;
  closed_count: number;
  total_realized_pnl: number;
}

export interface PaperTradeOpenRequest {
  symbol: string;
  type: TradeType;
  size: number;
  entry_price?: number;
  stop_loss?: number;
  take_profit?: number;
}
